low-level/files/opt/seagate/sspl/bin/generate_resource_health_view/resource_health_view

```diff
@@ -905,27 +908,66 @@ class SSPLHealthView(object):
         utility_instance = tool_factory.get_instance('sysfs')
         utility_instance.initialize()
         sas_ports_dict = utility_instance.get_phy_negotiated_link_rate()
+        sas_phy_dict = {}
         if sas_ports_dict:
             for key, value in sas_ports_dict.items():
                 link_rate = value.strip()
-                phy_number = key.split(":")[1]
-                resource_id = self.SAS_RESOURCE_ID + ':' + "phy-" + phy_number
+                phy_number = int(key.split(":")[1])
                 if 'Gbit'.lower() in link_rate.lower():
                     health = "OK"
-                elif 'Unknown'.lower() in link_rate.lower():
-                    health = "NA"
                 else:
                     health = "Fault"
+                sas_phy_dict.update({phy_number:health})
 
-                sas_port_data.update({
-                    resource_type+'-'+resource_id:{
-                        "alert_type": "NA",
-                        "severity": "NA",
+            # For SASHBA-0
+            if 'OK' in sas_phy_dict.values():
+                sas_hba_health = "OK"
+                sas_hba_severity = "NA"
+                sas_hba_alert_type = "NA"
+            else:
+                sas_hba_health = "Fault"
+                sas_hba_severity = "critical"
+                sas_hba_alert_type = "fault"
+
+            sas_port_data.update({
+                'sas_hbas':{
+                    resource_type+'-'+self.SAS_RESOURCE_ID:{
+                        "alert_type": sas_hba_alert_type,
+                        "severity": sas_hba_severity,
                         "alert_uuid": "NA",
                         "durable_id": "NA",
-                        "health": health,
+                        "health": sas_hba_health,
                         "fetch_time" : int(time.time())
-                    }})
+                    }
+                },
+                'sas_ports':{}
+            })
+
+            # For SAS Port
+            phy_lst_tpl = sorted([(k, v) for k, v in sas_phy_dict.items()])
+            phy_hlt = [phy[1] for phy in phy_lst_tpl]
+            # Make group of 4 phy for every port
+            phy_group = [phy_hlt[n:n+4] for n in range(0, len(phy_hlt), 4)]
```

Comment:
We will improve this hard-coding through the improvement tkt that we need to create.